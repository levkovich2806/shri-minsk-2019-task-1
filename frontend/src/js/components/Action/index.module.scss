.action {
  font-family: Helvetica, Arial, sans-serif;
  text-decoration: none;
  align-self: center;
  text-align: center;
  margin-bottom: 20px;
  color: rgba(51, 51, 51, 0.6);
  width: 100%;
  cursor: pointer;

  &__active {
    color: #333;
  }

  &__btn {
    background-color: #fc0;
    border-radius: 4px;
    padding: 10px 15px;
    color: #333;
    transition: background-color 0.5s;
  }

  &__btn:hover {
    background-color: #ffd735;
  }

  &__btn:active {
    background-color: #ffe993;
  }
}

.active {
  display: flex;
  position: absolute;
  top: 65px;
  background-color: white;
  z-index: 999;
  width: calc(100% - 40px);
  padding: 20px;
}

@media (min-width: 1100px) {
  .action {
    margin-left: 32px;
    margin-top: 18px;
  }
}
