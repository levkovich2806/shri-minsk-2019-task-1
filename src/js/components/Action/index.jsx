import React, { PureComponent } from "react";
import classnames from "classnames";
import PropTypes from "prop-types";

import styles from "./index.module.scss";

class Action extends PureComponent {
  render() {
    const { title, status, onClick } = this.props;
    return (
      <span
        onClick={onClick}
        className={classnames(
          styles.action,
          status && styles[`action__${status}`]
        )}
      >
        {title}
      </span>
    );
  }
}

Action.propTypes = {
  title: PropTypes.string.isRequired,
  status: PropTypes.string,
  onClick: PropTypes.func
};

Action.defaultProps = {
  title: "",
  status: "normal",
  onClick: () => Promise.resolve()
};

export default Action;
