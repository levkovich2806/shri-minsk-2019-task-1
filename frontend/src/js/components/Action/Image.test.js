import React from 'react';
import { shallow } from 'enzyme';

import Action from './index';

describe('Action', () => {

  it('Should match snapshot Action', () => {
    const wrapper = shallow(<Action title="Добавить" status="btn" onClick={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
