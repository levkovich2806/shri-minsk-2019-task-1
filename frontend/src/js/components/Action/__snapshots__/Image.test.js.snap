// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Should match snapshot Action 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Action
    onClick={[Function]}
    status="btn"
    title="Добавить"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Добавить",
      "className": "action action__btn",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Добавить",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Добавить",
        "className": "action action__btn",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Добавить",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
